docs/guides/go-built-in.md

```diff
@@ -462,16 +460,29 @@ execution of the block once accepted.
 In the following code, the application simply returns the unmodified group of transactions:
 
 ```go
-func (app *KVStoreApplication) PrepareProposal(_ context.Context, proposal *abcitypes.RequestPrepareProposal) (*abcitypes.ResponsePrepareProposal, error) {
-    return &abcitypes.ResponsePrepareProposal{Txs: proposal.Txs}, nil
-}
-```
+ func (app *KVStoreApplication) PrepareProposal(_ context.Context, proposal *abcitypes.RequestPrepareProposal) (*abcitypes.ResponsePrepareProposal, error) {
+   totalBytes := int64(0)
+   txs := make([]byte, 0)
+
+   for _, tx := range proposal.Txs {
+     totalBytes += int64(len(tx))
+     txs = append(txs, tx...)
 
-Once a proposed block is received by a node, the proposal is passed to the application to give
-its blessing before voting to accept the proposal.
+       if totalBytes > int64(proposal.MaxTxBytes) {
+         break
+       }
+     }
+
+     return &abcitypes.ResponsePrepareProposal{Txs: proposal.Txs}, nil
+ }
+ ```
+
+ Once a proposed block is received by a node, the proposal is passed to the application to determine its validity before voting to accept the proposal.
```

Comment:
This paragraph refers the to snipped below, that is, to processProposal, not to prepareProposal.
However, the following 2 paragraphs, which you added/edited, refer to prepareProposal, not processProposal.
You need to reorder these paragraphs.

Also, you removed the comment about processProposal being used to validate the proposed block; I think you should bring it back.